SeminarWeek2_Assignment/SeminarWeek2_Assignment/SeminarWeek2_Assignment/HelloViewController.swift

```diff
@@ -0,0 +1,106 @@
+
+    private lazy var email = "skydh1214@naver.com"
+
+    private lazy var tvingImageView: UIImageView = {
+        let welcomeImageView = UIImageView()
+        welcomeImageView.backgroundColor = .black
+        welcomeImageView.image = UIImage(named: "tvingimage")
+        return welcomeImageView
+    }()
+    
+    lazy var emailLabel: UILabel = {
+        let label = UILabel()
+        label.text = "\(String(describing: email))님"
+        label.textAlignment = .center
+        label.font = UIFont(name:"Pretendard-Bold", size: 23)
+        label.textColor = .tvingGray1
+        return label
+    }()
+    
+    private lazy var welcomeLabel: UILabel = {
+        let label = UILabel()
+        label.text = "반가워요!"
+        label.textAlignment = .center
+        label.font = UIFont(name:"Pretendard-Bold", size: 23)
+        label.textColor = .tvingGray1
+        return label
+    }()
+    
+    private lazy var goMainButton: UIButton = {
+        let button = UIButton()
+        button.setTitle("메인으로", for: .normal)
+        button.backgroundColor = UIColor(named: "red")
+        button.titleLabel?.font = UIFont.systemFont(ofSize: 14)
+        button.titleLabel?.textColor = UIColor .white
+        button.titleLabel?.textAlignment = .center
+        button.layer.cornerRadius = 3
+        button.addTarget(self, action: #selector(goMainButtonTapped), for: .touchUpInside)
+        return button
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setStyle()
+        setLayout()
+    }
+}
+
+
+// MARK: - HelloViewController 관련 속성(배경 및 레이아웃)
+extension HelloViewController {
+    private func setStyle() {
+        view.backgroundColor = .black
+    }
+    private func setLayout() {
+        view.addSubviews(tvingImageView,emailLabel,welcomeLabel,goMainButton)
+        
+        tvingImageView.snp.makeConstraints { make in
+            make.top.equalToSuperview().inset(58)
+            make.leading.trailing.equalToSuperview()
+            make.width.equalTo(375)
+            make.height.equalTo(210.94)
+        }
+        
+        emailLabel.snp.makeConstraints { make in
+            make.top.equalTo(tvingImageView.snp.bottom).offset(67.06)
+            make.centerX.equalToSuperview()
+        }
+        
+        welcomeLabel.snp.makeConstraints { make in
+            make.top.equalTo(tvingImageView.snp.bottom).offset(100)
+            make.centerX.equalToSuperview()
+        }
+        
+        goMainButton.snp.makeConstraints { make in
+            make.top.equalTo(emailLabel.snp.bottom).offset(284)
+            make.centerX.equalToSuperview()
+            make.width.equalTo(335)
+            make.height.equalTo(52)
+        }
+    }
+    
+    func backToStartViewController() {
+        let startViewController = StartViewController()
+       startViewController.modalPresentationStyle = .fullScreen
+        self.present(startViewController, animated: true)
+    }
```

Comment:
희희 이거 말하고 싶었는데 아직 아무도 리뷰 안남겨둬서 다행이닿ㅎㅎ

지금 아마 StartVC -> HelloVC 로 갈 때 present 하는 것 같은데,
또 HelloVC -> StartVC 로 갈 때 present 를 한다면, 
이전에 있던 StartVC 위에 HelloVC를 올리고, 그 위에 또 StartVC를 올리는 식으로 진행됩니다

위 방식대로 진행하면, 메모리에 올라간 VC 들은 deinit, 즉 메모리에서 해제되지 않고, 계속 아래 어딘가에 올라가있는 상태가 되는거죠

잘 설명이 됐는지는 모르겠지만,,, 자세한 설명은 아래 링크 보면 좋을 것 같아요!
https://baked-corn.tistory.com/132

요지는 HelloVC 에서 StartVC 로 돌아갈 때 새로운 VC 를 present 하는 것이 아니라,
(현재는 present 이므로) dismiss 함수를 불러줘야 한다는 것입니다 :)

고로 이렇게 코드를 바꾸면 좋을 것 같네요
```suggestion
    func backToStartViewController() {
        dismiss(animated: true)
    }
```

Reply:
참고로 메모리에서 사라졌는지는
deinit 를 사용하면 된답니다!
```swift
deinit {
    // 무언가 deinit 되었다는걸 알 수 있는 표시?
    print("deinit : \(self)")
}
```

Reply:
오!! 감사합니당 오늘도 배우고 갑니다!!